framework/util/image_writer.cpp

```diff
@@ -96,6 +97,15 @@ static size_t                     temporary_buffer_size = 0;
         }                                                                                             \
     }
 
+static void ResizeTemoraryBuffer(size_t new_size)
```

Comment:
Could the temporary be `vector<uint8_t>` and then you can use `resize`?

Reply:
That is to say, you could call `resize` instead of calling this function.

Reply:
sure